code/iaas/logic/src/main/resources/META-INF/cattle/core-process/spring-core-processes-context.xml

```diff
@@ -144,4 +144,23 @@
     <!-- Cluster host map -->
     <process:process name="clusterhostmap.create" resourceType="clusterHostMap" start="requested" transitioning="creating" done="created" />
     <process:process name="clusterhostmap.remove" resourceType="clusterHostMap" start="created,creating" transitioning="removing" done="removed" />
+
+    <!-- Service Discovery Environment -->
+    <process:process name="environment.create" resourceType="environment" start="requested" transitioning="activating" done="active" />
+    <process:process name="environment.remove" resourceType="environment" start="active, activating" transitioning="removing" done="removed" />
+    
+    <!-- Service Discovery Service -->
+    <process:process name="service.create" resourceType="service" start="requested" transitioning="registering" done="inactive" />
+    <process:process name="service.activate" resourceType="service" start="inactive" transitioning="activating" done="active" />
+    <process:process name="service.deactivate" resourceType="service" start="active,activating" transitioning="deactivating" done="inactive" />
+    <process:process name="service.remove" resourceType="service" start="inactive, registering, active,activating" transitioning="removing" done="removed" />
+    
+    <!-- Service Discovery Service/Instance map -->
+    <process:process name="serviceexposemap.create" resourceType="serviceExposeMap" start="requested" transitioning="activating" done="active" />
```

Comment:
Do the resources need the word map in them? Does `serviceExpose` and `serviceConsume` sound any better?

Just a suggestion, you can take it or leave it.